internal/pkg/deploy/cloudformation/pipeline.go

```diff
@@ -36,8 +38,12 @@ func (cf CloudFormation) PipelineExists(in *deploy.CreatePipelineInput) (bool, e
 }
 
 // CreatePipeline sets up a new CodePipeline for deploying services.
-func (cf CloudFormation) CreatePipeline(in *deploy.CreatePipelineInput) error {
-	s, err := toStack(stack.NewPipelineStackConfig(in))
+func (cf CloudFormation) CreatePipeline(in *deploy.CreatePipelineInput, bucketName string) error {
+	templateURL, err := cf.pushTemplateToS3Bucket(bucketName, stack.NewPipelineStackConfig(in))
```

Comment:
For my own edification, are we always pushing the template to S3 regardless of size? Just wondering if we'd considered only doing that for templates that exceeded a certain size, just to save a roundtrip to S3 downstream.

Reply:
Do we tend to have conventions on adding function arguments to existing exported functions? I know that `CreatePipeline` and `UpdatePipeline` are only called internally from other packages, but was wondering if we might consider adding the `bucketName` to `deploy.CreatePipelineInput` within `Execute`, which would save on having to add the argument downstream.

Reply:
> Do we tend to have conventions on adding function arguments to existing exported functions? I know that `CreatePipeline` and `UpdatePipeline` are only called internally from other packages, but was wondering if we might consider adding the `bucketName` to `deploy.CreatePipelineInput` within `Execute`, which would save on having to add the argument downstream.

This did cross my mind (and reminded me of how to pass `shouldOutputJSON` with `env_show` a million years ago)! It does seem cleaner to add it to the input struct; the thing that stopped me was the consideration that `bucketName` isn't strictly "pipeline input." I can make this change, especially if we decide to push only too-big templates!

Reply:
> For my own edification, are we always pushing the template to S3 regardless of size? Just wondering if we'd considered only doing that for templates that exceeded a certain size, just to save a roundtrip to S3 downstream.

Yup, all templates will get pushed. To keep things simple and bc S3 is cheap! :)